### PR TITLE
First pass at Bean Validation

### PR DESCRIPTION
**Pull request by ggam**
Same comments as in #148

### CODE REVIEW

**Comment by arjantijms** [comment]
> Jakarta Server Faces

This should be "Jakarta Faces". Because the Mojarra project included the API this hasn't been so clear, but see e.g. https://github.com/eclipse-ee4j/faces-api

**Comment by ggam** [comment]
I knew this was the original intention but I was told it was finally named
Jakarta Server Faces. We have used that name on the Platform spec document.
Is Jakarta Faces the short name?

Btw, I have to restore CDI references now we are sure we can use that
acronym.

**Comment by arjantijms** [comment]
Hi,

It should have been Jakarta Faces as full name. Not sure where “Jakarta
Server Faces” was decided.

Kind regards,
Arjan

**Comment by ggam** [comment]
I remember reading ir from you but I thought it was kept as Server Faces
for consistency with Server Pages. IMO Jakarta Faces and Jakarta Pages
would have been better names.

**Comment by arjantijms** [comment]
>  IMO Jakarta Faces and Jakarta Pages would have been better names.

In case of Jakarta faces, the package name is now "javax.faces.*", and will become "jakarta.faces.*", so it matches perfectly. "server" is actually rarely if ever used in the API. It's always "FacesServlet", "FacesContext", etc.

It's also "OmniFaces" and "PrimeFaces", not "OmniServerFaces" and "PrimeServerFaces" :P

**Comment by bshannon** [comment]
For whatever reason, "Server" ended up in the name.  Maybe it will be renamed in
the future, but for now please update the name throughout.

File: src/main/jbake/content/bean-validation.adoc [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
 ~~~~~~
-= Introduction to Bean Validation
+= Introduction to Jakarta Bean Validation
 
 
 [[CHDGJIIA]][[introduction-to-bean-validation]]
 
-23 Introduction to Bean Validation
+23 Introduction to Jakarta Bean Validation
 ----------------------------------
```
Comment (bshannon) [inline src/main/jbake/content/bean-validation.adoc]: Fix underline length.

File: src/main/jbake/content/bean-validation002.adoc [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 ~~~~~~
-# Using Bean Validation Constraints
+# Using Jakarta Bean Validation Constraints
 
 
 [[GIRCZ]][[using-bean-validation-constraints]]
 
-Using Bean Validation Constraints
+Using Jakarta Bean Validation Constraints
 ---------------------------------
```
Comment (bshannon) [inline src/main/jbake/content/bean-validation002.adoc]: Fix underline length.

File: src/main/jbake/content/bean-validation005.adoc [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
 ~~~~~~
-Further Information about Bean Validation
+Further Information about Jakarta Bean Validation
 =========================================
 
 [[CACDECFE]][[further-information-about-bean-validation]]
 
-Further Information about Bean Validation
+Further Information about Jakarta Bean Validation
 -----------------------------------------
```
Comment (bshannon) [inline src/main/jbake/content/bean-validation005.adoc]: length

File: src/main/jbake/content/bean-validation005.adoc [hunk 4]
```diff
@@ -1,20 +1,20 @@
 type=page
 status=published
-title=Further Information about Bean Validation
+title=Further Information about Jakarta Bean Validation
 next=bean-validation-advanced.html
 prev=bean-validation004.html
 ~~~~~~
-Further Information about Bean Validation
+Further Information about Jakarta Bean Validation
 =========================================
```
Comment (bshannon) [inline src/main/jbake/content/bean-validation005.adoc]: length

File: src/main/jbake/content/bean-validation005.adoc [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
+Further Information about Jakarta Bean Validation
 -----------------------------------------
 
-For more information on Bean Validation, see
+For more information on Jakarta Bean Validation, see
 
-* Bean Validation 2.0 Specification:
+* Jakarta Bean Validation 2.0 Specification:
 +
 `http://www.jcp.org/en/jsr/detail?id=380`
```
Comment (bshannon) [inline src/main/jbake/content/bean-validation005.adoc]: https://jakarta.ee/specifications/bean-validation/

**Comment by ggam** [comment]
> For whatever reason, "Server" ended up in the name. Maybe it will be renamed in
> the future, but for now please update the name throughout.

Haven't read this comment yet. I'll update it everywhere then.

**Pull request merged**